### DISCUSSION TITLE
How to get role name

### DISCUSSION THREAD

**Discussion by luis9430** [Power-Components/livewire-powergrid#239] [Q&A] [answered]
Hi, I'm using the roles spatie package, I'm trying to get the role name for each user, how should I do it?

in datasource I have this

![image](https://user-images.githubusercontent.com/30971535/150694787-34b58f47-2a73-4454-8ea9-d42dba40d783.png)

 but it returns an array with all the data
![image](https://user-images.githubusercontent.com/30971535/150694732-1ab4ba35-10fc-4b61-b47e-fca5eb753437.png)

I hope you can help me, thanks

**Comment by dansysanalyst** [comment, marked as answer]
Hi Luis,

I am not sure if this is the best approach, I believe it's not enough for filtering by the relationship for example.

I have never used Spatie Roles, but I had a go with it:

```php
    public function datasource(): ?Builder
    {
        return User::query()->with('roles');
    }
```
It's unclear to me if a user can have multiple roles. 
 
A`dd($user->roles)` shows me:

<img width="376" alt="CleanShot 2022-01-23 at 23 39 38@2x" src="https://user-images.githubusercontent.com/79267265/150700868-57d70a17-25e2-4929-b925-f6ffeb02b534.png">

So, I follow this relationship, plucking the Roles names and joining them together by ",".

```php
    public function addColumns(): ?PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            ->addColumn('last_name')
            ->addColumn('role_name', function (User $user) {
                return $user->roles
                    ->pluck('name')
                    ->implode(',');
            });
    }
```

And I get:

<img width="1147" alt="CleanShot 2022-01-23 at 23 26 53@2x" src="https://user-images.githubusercontent.com/79267265/150700737-261a8c1c-9e5c-488e-99e9-c1258b834a1c.png">

@luanfreitasdev a better idea would be to Join table?

https://livewire-powergrid.com/#/table/datasource?id=join-tables

**Comment by luis9430** [comment]
hello, thank you very much for your answer, I tried it and it worked
 and yes, users can have multiple roles that returns an array... 
and about filtering the information I will investigate how it can be done, thanks!

**Reply by LinceIberico** [reply]
Hello everyone, have you found anything else? I have the same problem and I can't find the solution. I can't get role filtering to work.

With this function the roles appear but it does not filter either, the error is always the same because it makes a sql query in the filters.
                ->makeInputSelect(Role::all(),'name','id', ['live-search' => true]),
